mewarpx/medocker/docker_shared/run_warpx.sh

```diff
@@ -119,21 +156,30 @@ do
     find . -type f -exec touch -a -c {} +
 done
 }
+
 # https://stackoverflow.com/questions/21465297/tee-stdout-and-stderr-to-separate-files-while-retaining-them-on-their-respective
-{ { bash $SCRIPTNAME; } > >( tee -a stdout.out ); } \
+bash $SCRIPTNAME > >( tee -a stdout.out ) \
                        2> >( tee -a stderr.err >&2 ) &
 
+# sleep so that there is time for the mpiexec process to launch
+sleep 2
+
 MAINID=$!
+MPIEXECID=$(pgrep -P $MAINID -f mpiexec)
+echo "main process id is $MAINID"
+echo "mpiexec process id: $MPIEXECID"
 checkpoint &
 CHECKID=$!
-wait -n $MAINID $CHECKID
+pingterminatewarning &
+PINGID=$!
+wait $MAINID
 EXITFLAG=$?
 # kill, but silently, as one will error no matter what. It looks like killing
 # $MAINID actually isn't enough -- child processes still run -- but it
 # shouldn't matter as the docker container should exit when this script exits,
 # I think.
 kill $CHECKID &> /dev/null
-kill $MAINID &> /dev/null
+kill $PINGID &> /dev/null
```

Comment:
Based on my above comment, keep the $MAINID kill here I believe.

Reply:
I guess it should kill `$MPIEXECID` though (which would anyway finish `$MAINID`). I'll use 
```suggestion
kill $PINGID &> /dev/null
kill -KILL "${MPIEXECID}" &> /dev/null
```
to avoid WarpX trying to write another checkpoint (which would presumably not complete in time).